public/app/core/utils/richHistory.ts

```diff
@@ -229,30 +207,21 @@ export function mapQueriesToHeadings(query: RichHistoryQuery[], sortOrder: SortO
   return mappedQueriesToHeadings;
 }
 
-/* Create datasource list with images. If specific datasource retrieved from Rich history is not part of
- * exploreDatasources add generic datasource image and add property isRemoved = true.
+/*
+ * Create a list of all available data sources
  */
-export function createDatasourcesList(queriesDatasources: string[]) {
-  const datasources: Array<{ label: string; value: string; imgUrl: string; isRemoved: boolean }> = [];
+export function createDatasourcesList() {
+  const datasources: Array<{ name: string; imgUrl: string; uid: string }> = [];
 
-  queriesDatasources.forEach((dsName) => {
-    const dsSettings = getDataSourceSrv().getInstanceSettings(dsName);
-    if (dsSettings) {
+  getDataSourceSrv()
+    .getList()
+    .forEach((dsSettings) => {
       datasources.push({
-        label: dsSettings.name,
-        value: dsSettings.name,
+        name: dsSettings.name,
+        uid: dsSettings.uid,
         imgUrl: dsSettings.meta.info.logos.small,
-        isRemoved: false,
       });
-    } else {
-      datasources.push({
-        label: dsName,
-        value: dsName,
-        imgUrl: 'public/img/icn-datasource.svg',
-        isRemoved: true,
-      });
-    }
-  });
+    });
```

Comment:
super tiny nit: i think it would be easier to read with something like
```
return getDataSourceSrv()
    .getList()
    .map(
    ...
    )
```

(but no need to change)

Reply:
Good point! Updated in 249ea79b378fcc54f3bf4cb6db9941a0ef8d1b0e